timeout while waiting for state to become 'success' (timeout: 2m0s) #760

There was a patch predicted to address the timeouts but it does not appear to have been successful. 

The PR, https://github.com/PagerDuty/terraform-provider-pagerduty/pull/763,  seems to address the issue by making changes to the **EVENTS** API but the timeout we suspect is in the **REST** API (reference: https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTUz-rate-limiting)

For full issue history, see https://github.com/PagerDuty/terraform-provider-pagerduty/issues/760

seems like it might be related to how the terraform SDK handles retries [[related issue]](https://github.com/hashicorp/terraform-plugin-sdk/issues/530)

I'm receiving the same WaitForState errors the users in the issue above (and related issues) are seeing:

```
2023-11-27T22:16:04.626Z [WARN]  provider.terraform-provider-pagerduty_v3.1.2: WaitForState timeout after 2m0s: timestamp=2023-11-27T22:16:04.626Z
2023-11-27T22:16:04.626Z [WARN]  provider.terraform-provider-pagerduty_v3.1.2: WaitForState starting 30s refresh grace period: timestamp=2023-11-27T22:16:04.626Z
2023-11-27T22:16:34.628Z [ERROR] provider.terraform-provider-pagerduty_v3.1.2: WaitForState exceeded refresh grace period: timestamp=2023-11-27T22:16:34.626Z
2023-11-27T22:16:34.628Z [ERROR] vertex "module.<pagerduty_service_module>" error: timeout while waiting for state to become 'success' (timeout: 2m0s)
2023-11-27T22:16:34.628Z [ERROR] vertex "module.<pagerduty_service_integration> (expand)" error: timeout while waiting for state to become 'success' (timeout: 2m0s)
```

--- a/go.mod
+++ b/go.mod
@@ -9,7 +9,7 @@ require (
 	github.com/hashicorp/terraform-exec v0.16.0
 	github.com/hashicorp/terraform-json v0.13.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.11.0
-	github.com/heimweh/go-pagerduty v0.0.0-20231201163338-a11b2f79dfcf
+	github.com/heimweh/go-pagerduty v0.0.0-20231201203054-09be11e40bea
 )
 
 require (

--- a/pagerduty/resource_pagerduty_service.go
+++ b/pagerduty/resource_pagerduty_service.go
@@ -480,7 +480,6 @@ func fetchService(d *schema.ResourceData, meta interface{}, errCallback func(err
 
 			errResp := errCallback(err, d)
 			if errResp != nil {
-				time.Sleep(2 * time.Second)
 				return resource.RetryableError(errResp)
 			}
 

--- a/pagerduty/resource_pagerduty_service_integration.go
+++ b/pagerduty/resource_pagerduty_service_integration.go
@@ -635,7 +635,6 @@ func fetchPagerDutyServiceIntegration(d *schema.ResourceData, meta interface{}, 
 
 			errResp := errCallback(err, d)
 			if errResp != nil {
-				time.Sleep(2 * time.Second)
 				return resource.RetryableError(errResp)
 			}
 
@@ -732,7 +731,6 @@ func resourcePagerDutyServiceIntegrationCreate(d *schema.ResourceData, meta inte
 	retryErr := resource.Retry(1*time.Minute, func() *resource.RetryError {
 		if serviceIntegration, _, err := client.Services.CreateIntegration(service, serviceIntegration); err != nil {
 			if isErrCode(err, 400) {
-				time.Sleep(2 * time.Second)
 				return resource.RetryableError(err)
 			}
 

